### PR TITLE
Cli: get terraform binary from inside the package

### PR DESCRIPTION
**Pull request by trawler**
This PR resolves an issue whereby the tectonic binary didn't necessarily use the packages terraform binary

### CODE REVIEW

**Comment by trawler** [comment]
ok to test

File: installer/pkg/workflow/executor.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+//
+// An error is returned if the TerraForm binary could not be found, or if the
+// TerraForm call itself failed, in which case, details can be found in the
+// output.
+func (ex *executor) execute(clusterDir string, args ...string) error {
+	// Prepare TerraForm command by setting up the command, configuration,
+	// and the working directory
+	cmd := exec.Command(ex.binaryPath, args...)
+
+	if clusterDir == "" {
```
Comment (enxebre) [inline installer/pkg/workflow/executor.go]: small hint: this should probably go at the top of the func before creating the cmd struct

File: installer/pkg/workflow/executor.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+var ErrBinaryNotFound = errors.New(
+	"TerraForm not in executable's folder, cwd nor PATH",
+)
+
+// newExecutor initializes a new Executor.
+func newExecutor() (*executor, error) {
+	ex := new(executor)
+
+	// Find the TerraForm binary.
+	out, err := tfBinaryPath()
```
Comment (enxebre) [inline installer/pkg/workflow/executor.go]: `out` sounds a bit confusing to me, I'd rather use `bin` or `path`, wdyt?

File: installer/pkg/workflow/executor.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+// Set the binary names for different platforms
+const (
+	tfBinUnix    = "terraform"
+	tfBinWindows = "terraform.exe"
+)
+
+// ErrBinaryNotFound denotes the fact that the TerraForm binary could not be
+// found on disk.
+var ErrBinaryNotFound = errors.New(
```
Comment (enxebre) [inline installer/pkg/workflow/executor.go]: no need to export this variable

File: installer/pkg/workflow/executor.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+// executor enables calling TerraForm from Go, across platforms, with any
+// additional providers/provisioners that the currently executing binary
+// exposes.
+//
+// The TerraForm binary is expected to be in the executing binary's folder, in
+// the current working directory or in the PATH.
+type executor struct {
+	binaryPath   string
+	envVariables map[string]string
```
Comment (enxebre) [inline installer/pkg/workflow/executor.go]: this is no used anywhere

File: installer/pkg/workflow/executor.go [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+	cmd := exec.Command(ex.binaryPath, args...)
+
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	cmd.Dir = clusterDir
+
+	// Start TerraForm.
+	err := cmd.Run()
```
Comment (enxebre) [inline installer/pkg/workflow/executor.go]: you can just:
`return cmd.Run()` here and avoid the `if err` verbosity

**Pull request merged**